File: backend/src/main/java/com/ody/meeting/dto/response/MateResponse.java
package com.ody.meeting.dto.response;

import com.ody.mate.domain.Mate;
import java.util.List;

public record MateResponse(String nickname) {

    public static List<MateResponse> from(List<Mate> mates) {
        return mates.stream()
                .map(mate -> new MateResponse(mate.getNickname().getNickname()))
                .toList();
    }
}

File: android/app/src/main/java/com/woowacourse/ody/data/remote/core/entity/meeting/response/MateResponse.kt
package com.woowacourse.ody.data.remote.core.entity.meeting.response

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.woowacourse.ody.domain.model.Mate

@JsonClass(generateAdapter = true)
data class MateResponse(
    @Json(name = "nickname")
    val nickname: String,
) {
    fun toMate(): Mate = Mate(nickname = nickname)
}
